### ISSUE TITLE
PHP Fatal Error due to endurance-page-cache.php

### ISSUE THREAD

**Issue by borreLore** [bluehost/endurance-page-cache#40]
Hello, I am having a weird error suddenly pop up on my site that appears to be due to the endurance-page-cache. From my research, it looks like a lot of people have been opening posts on here and getting the issue resolved. In some cases, it even looks like the person who appears to be the creator of this script, MikeHansenMe, drops by to help out. 

The full error is: 

Fatal error: Call to undefined function is_user_logged_in() in /home1/borrelore/public_html/nova-technica.com/wp-content/mu-plugins/endurance-page-cache.php on line 370

The first instance of this error according to the error log on my site appears to be [07-Jul-2018 04:33:22 UTC]. I am pretty new to running a website, and unfortunately, I don't know enough about this script to know why it's even on my site, or why it took my website down. Is anybody able to give some tips, or provide some kind of steps users who find this script on their site can follow to resolve this issue? I am eager to resolve this issue to get my site back up and running. 

To whoever helps, thanks in advance!

**Comment by MikeHansenMe** [comment]
Hi @borreLore can you give me a domain name? I can see if it is something I can look into for you.

**Comment by borreLore** [comment]
Hi @MikeHansenMe , sure, no problem. My domain is http://nova-technica.com/

Thanks for the reply!

**Comment by borreLore** [comment]
Hi @MikeHansenMe , have you had a chance to see if you can spot what is causing this error? 

I have checked the PHP code, but unfortunately I have not worked with PHP before and it is difficult for me to troubleshoot this. All I know is that it suddenly stopped working on that date after working fine for the past year. My first guess was there was an update to the endurance-page-cache.php, but I don't have any way to verify if this is the case. The only other thing that comes to mind is that another addon had updated, and this conflicted with the endurance-page-cache.php code. Of course, as much of wordpress code is still pretty alien to me, I wouldn't be surprised if I'm wrong on all accounts.

**Comment by MikeHansenMe** [comment]
Hey @borreLore I took a quick look but I do not have access to that site (only Bluehost sites), Typically when an error similar to the one you are seeing occurs, it means something else is erroring and output buffering  (in this plugin) is catching it. What I would do to solve the problem is:

Rename endurance-page-cache.php to endurance-page-cache.php.bak
Reload the site
Take note of the new error being logged/displayed
Resolve that error
Rename endurance-page-cache.php.bak to endurance-page-cache.php
Everything should be functional again at this point.

**Comment by borreLore** [comment]
Hello @MikeHansenMe , awesome, thanks for those steps! I have renamed it in the past, and you are right, there is another error that pops up at that point. I will give it a shot to resolve this error. Thanks again for your help!

**Comment by borreLore** [comment]
Hello @MikeHansenMe and anybody else having this problem. I managed to resolve the issue, and I wanted to post my steps for anybody else experiencing this. 

1. I tried @MikeHansenMe 's suggestion to rename the endurance-page-cache.php file to endurance-page-cache.php.bak, but this resulted in the white screen of death. There was simply no error message that followed, so I renamed it back to the original and kept searching for a solution. 
2. At this point, I went into my file manager on my host site, and went into the regular **plugins** folder.
3. I tried @MikeHansenMe 's renaming the file trick on each plugin, one at a time, and refreshed the page. Eventually, a long error message popped up and referred to one of my plugins (Robo Gallery in my case) as no longer existing. Not sure if the file corrupted, or was out of date, but: 
4. I renamed the Robo Gallery with the .bak at the end, and removed the .bak from the other plugins. The site loaded successfully after this. 

So basically, my take away is that if all else fails, brute force the problem by renaming your plugins with a .bak at the end one at a time until the issue is fixed. 

Thanks @MikeHansenMe for your help on this, you pointed me in the right direction for the fix!
